### ISSUE TITLE
Next.js SSR and row-level security

### ISSUE THREAD

**Issue by gr-qft** [supabase/supabase#3834]
First off, thanks the team for this awesome library! 

I'm using magic link authentication. I have found that even after a user has been identified, and if row-level security is enabled (to allow only a user with a specific email address), then inside `getServerSideProps` the data cannot be fetched using supabase client, i.e. this code is returning empty data and null error (and the user has been identified)

```js
const { user } = await supabase.auth.api.getUserByCookie(req);

  if (!user) {
    return { props: {}, redirect: { destination: "/" } };
  }

  const { data, error } = await supabase.from(...)
    ...
```  
Hope you will help me see what's going on. 

Thanks!!

**Comment by kiwicopple** [comment]
Hey @gr-qft - you will still need to use `setAuth()` on the client: https://supabase.io/docs/reference/javascript/auth-setauth

After that you can make requests "as" the user

**Comment by gr-qft** [comment]
@kiwicopple Thanks for the hint. I'll try this and let you know. By the way, this makes sense, because if i turn off low-level security then the fetch will be successful.

**Comment by gr-qft** [comment]
@kiwicopple Thanks again to the team! Awesome work.

About the problem, I just don't know what to do to pass `access_token` to `getServerSideProps`; it seems `access_token` can only be obtained in the frontend (I tried `supabse.auth.session()` inside `getServerSideProps`as well  but it gave me `null`). And it's actually a bit strange that the user can be obtained correctly inside `getServerSideProps` and yet with row-level security setup the data still cannot be fetched. Is there a reason why **supabase** is not enabling that already? 

```js
export async function getServerSideProps({ req }: any) {
  const { user } = await supabase.auth.api.getUserByCookie(req);

  if (!user) {
    return { props: {}, redirect: { destination: "/" } };
  }

  const { data, error } = await supabase
    .from("")
    ... 
   
}
```
P.S. I prefer not having to do something like writing the 'access_token' to the database from the frontend, then fetch it again inside `getServerSideProps`, just to be able to set it using `setAuth()`. 

Thanks!

**Comment by kiwicopple** [comment]
I may not be understanding your question properly, so if not feel free to comment back

> I just don't know what to do to pass access_token to getServerSideProps

I assume that `getServerSideProps` is your own route? Ideally when you first authenticate the user you call [setAuthCookie](https://github.com/supabase/gotrue-js/blob/82f4bca4843f62e02da61d8217adb5b4350a1856/src/GoTrueApi.ts#L436) in a server route. This will create a cookie, which is then passed with all network requests.

After that, you can pull the user's details out of the cookie - you don't need to store `access_token` anywhere. I just updated the latest version of supabase-js to support getting the token directly from the cookie. It should work like this:

```
export async function getServerSideProps({ req }: any) {
  const { token, user } = await supabase.auth.api.getUserByCookie(req);

  if (!user) {
    return { props: {}, redirect: { destination: "/" } };
  }

  const { data, error } = await supabase.setAuth(token)
    .from("")
    ... 
   
}
```

I'll close this one for now, but if there is a bug that you feel needs fixing just let me know and I'll reopen

**Comment by gr-qft** [comment]
@kiwicopple Thanks a lot, supabase team is awesome! 

Just a short clarification: I did use `setAuthCookie` and that was why I was able to call `getUserByCookie` inside `getServerSideProps`. As I commented above, I could then get `user`, but the following call to fetch the tables (i.g.  `const { data, error } = await supabase.from("")...`) just return null data and undefined error when low-level security is set up. 

I'm checking this 'token' that you just made available and will let you know. Thanks again!
